explorer/src/providers/accounts/block.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import * as Sentry from "@sentry/react";
+import * as Cache from "providers/cache";
+import { RetrieveSlot, RetrieveBlockhash } from "@theronin/solarweave";
+import { useCluster, Cluster } from "../cluster";
+import { ActionType, FetchStatus } from "providers/cache";
+import { HistoryProvider } from "./history";
+import { TokensProvider } from "./tokens";
+
+type State = Cache.State<Block>;
+type Dispatch = Cache.Dispatch<Block>;
+
+const StateContext = React.createContext<State | undefined>(undefined);
+const DispatchContext = React.createContext<Dispatch | undefined>(undefined);
+
+type BlockProviderProps = { children: React.ReactNode };
+
+export interface Block {
+  BlockData: any;
+  Tags: any;
+}
+
+export function BlockProvider({ children }: BlockProviderProps) {
+  const { url } = useCluster();
+  const [state, dispatch] = Cache.useReducer<Block>(url);
+
+  React.useEffect(() => {
+    dispatch({ type: ActionType.Clear, url });
+  }, [dispatch, url]);
+
+  return (
+    <StateContext.Provider value={state}>
+      <DispatchContext.Provider value={dispatch}>
+        <TokensProvider>
+          <HistoryProvider>{children}</HistoryProvider>
+        </TokensProvider>
```

Comment:
please remove the token and history providers, I think they were copy pasted 😉

Reply:
Indeed they were hahaha.